Objects/stringlib/codecs.h

```diff
@@ -449,14 +449,14 @@ STRINGLIB(utf8_encoder)(PyObject *unicode,
     Py_XDECREF(error_handler_obj);
     Py_XDECREF(exc);
 #endif
-    return _PyBytesWriter_Finish(&writer, p);
+    return p;
 
 #if STRINGLIB_SIZEOF_CHAR > 1
  error:
     Py_XDECREF(rep);
     Py_XDECREF(error_handler_obj);
     Py_XDECREF(exc);
-    _PyBytesWriter_Dealloc(&writer);
+    _PyBytesWriter_Dealloc(writer);
```

Comment:
When you pass a writer to a function, I prefer that the caller is responsible to call _PyBytesWriter_Finish() and _PyBytesWriter_Dealloc(). Here it doesn't if it's success (non-NULL) or an error (NULL) :-(